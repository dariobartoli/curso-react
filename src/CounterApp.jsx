import React from 'react'
import PropTypes from 'prop-types'
import { useState } from 'react'

const CounterApp = ({value}) => {
    const [counter, setCounter] = useState(value)

    const handleAdd = () => {
        setCounter(counter + 1)
    }
    const handleSubstract = () => {
        setCounter(counter - 1)
    }
    const handleReset = () => {
        setCounter(value)
    }

  return (
    <>
        <p data-testid='test-value'>{counter}</p>
        <button onClick={handleAdd}>sumar +1</button>
        <button onClick={handleSubstract}>restar -1</button>
        <button aria-label='btn-reset' onClick={handleReset}>reset</button>
    </>
  )
}
CounterApp.propTypes = {
  value: PropTypes.number.isRequired,
}

export default CounterApp