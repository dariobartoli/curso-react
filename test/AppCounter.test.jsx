import { fireEvent, render,screen } from "@testing-library/react"
import CounterApp from "../src/CounterApp"

describe('Pruebas en <AppCounter/>', () => { 
    const value = 100
    test('debe hacer match con el snapshot', () => { 
        const { container } = render( <CounterApp value={value}/>)
        expect(container).toMatchSnapshot()
    })
    test('debe mostrar el valor inicial de 100', () => { 
        render( <CounterApp value={value}/>)
        expect(screen.getByText(value).innerHTML).toBe('100')
    })

    //simular eventos
    test('debe de incrementar con el boton +1', () => { 
        render( <CounterApp value={value}/>)
        fireEvent.click( screen.getByText('sumar +1') )
        expect( screen.getByText(value+1)).toBeTruthy();
    })
    test('debe de decrementar con el boton -1', () => { 
        render( <CounterApp value={value}/>)
        fireEvent.click( screen.getByText('restar -1') )
        expect( screen.getByText(value-1)).toBeTruthy();
    })
    test('debe de funcionar el reset', () => { 
        render( <CounterApp value={value}/>)
        fireEvent.click( screen.getByText('sumar +1') )
        fireEvent.click( screen.getByText('sumar +1') )
        fireEvent.click( screen.getByText('sumar +1') )
        fireEvent.click( screen.getByText('sumar +1') )
        fireEvent.click( screen.getByRole('button', {name: 'btn-reset'})) //otra forma de apuntar a un elemento es con el aria-label, que cambia el valor del name

        expect( screen.getByText(value)).toBeTruthy()
    })
})