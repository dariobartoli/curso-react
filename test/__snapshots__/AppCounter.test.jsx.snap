// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AppCounter/> debe hacer match con el snapshot 1`] = `
<div>
  <p
    data-testid="test-value"
  >
    100
  </p>
  <button>
    sumar +1
  </button>
  <button>
    restar -1
  </button>
  <button
    aria-label="btn-reset"
  >
    reset
  </button>
</div>
`;
